sdk/containerregistry/container-registry/review/container-registry.api.md

```diff
@@ -208,7 +208,7 @@ export interface RegistryArtifact {
     listTagProperties(options?: ListTagPropertiesOptions): PagedAsyncIterableIterator<ArtifactTagProperties>;
     readonly registryEndpoint: string;
     readonly repositoryName: string;
-    updateManifestProperties(options?: UpdateManifestPropertiesOptions): Promise<ArtifactManifestProperties>;
+    updateManifestProperties(options: UpdateManifestPropertiesOptions): Promise<ArtifactManifestProperties>;
```

Comment:
Why are we making this non-optional when all of the keys are optional? Do one of the props have to be true?

Reply:
It doesn't make sense to not pass anything to the update method. The properties are optional to updating one of them without having to specify all of them.  Is it better to make it optional then no-op when nothing/empty object is passed in?

Reply:
Yeah, I think you're right to make this required, just as a hint that they should do something. 

As an aside, I spent some time getting TS magic for 'at least one of these properties should be on this object':

```ts
type AtLeastOne<Type> =
    {
        [Property in keyof Type]: Required<Pick<Type, Property>> & Partial<Type>
    }[keyof Type]
```

It works and does the right thing, but I think it's overkill for this. Maybe it'll be useful in a different place.